epochStart/bootstrap/syncValidatorStatus.go

```diff
@@ -0,0 +1,165 @@
+	s := &syncValidatorStatus{
+		dataPool:       args.DataPool,
+		marshalizer:    args.Marshalizer,
+		requestHandler: args.RequestHandler,
+	}
+	syncMiniBlocksArgs := sync.ArgsNewPendingMiniBlocksSyncer{
+		Storage:        &disabled.Storer{},
+		Cache:          s.dataPool.MiniBlocks(),
+		Marshalizer:    s.marshalizer,
+		RequestHandler: s.requestHandler,
+	}
+	var err error
+	s.miniBlocksSyncer, err = sync.NewPendingMiniBlocksSyncer(syncMiniBlocksArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	return s, nil
+}
+
+// NodesConfigFromMetaBlock synces and creates registry from epoch start metablock
+func (s *syncValidatorStatus) NodesConfigFromMetaBlock(
+	currMetaBlock *block.MetaBlock,
+	prevMetaBlock *block.MetaBlock,
+	publicKey []byte,
+) (*sharding.NodesCoordinatorRegistry, uint32, error) {
+	if !currMetaBlock.IsStartOfEpochBlock() {
+		return nil, 0, epochStart.ErrNotEpochStartBlock
+	}
+	if !prevMetaBlock.IsStartOfEpochBlock() {
+		return nil, 0, epochStart.ErrNotEpochStartBlock
+	}
+
+	nodesConfig := &sharding.NodesCoordinatorRegistry{
+		EpochsConfig: make(map[string]*sharding.EpochValidators),
+		CurrentEpoch: currMetaBlock.Epoch,
+	}
+
+	epochValidators, selfShardId, err := s.processNodesConfigFor(currMetaBlock, publicKey)
+	if err != nil {
+		return nil, 0, err
+	}
+	configId := fmt.Sprint(currMetaBlock.Epoch)
+	nodesConfig.EpochsConfig[configId] = epochValidators
+
+	epochValidators, _, err = s.processNodesConfigFor(prevMetaBlock, nil)
+	if err != nil {
+		return nil, 0, err
+	}
+	configId = fmt.Sprint(prevMetaBlock.Epoch)
+	nodesConfig.EpochsConfig[configId] = epochValidators
+
+	// TODO: use shuffling process from nodesCoordinator to create the final data
+
+	return nodesConfig, selfShardId, nil
+}
+
+func (s *syncValidatorStatus) processNodesConfigFor(
+	metaBlock *block.MetaBlock,
+	publicKey []byte,
+) (*sharding.EpochValidators, uint32, error) {
+	shardMBHeaders := make([]block.ShardMiniBlockHeader, 0)
+	for _, mbHeader := range metaBlock.MiniBlockHeaders {
+		if mbHeader.Type != block.PeerBlock {
+			continue
+		}
+
+		shardMBHdr := block.ShardMiniBlockHeader{
+			Hash:            mbHeader.Hash,
+			ReceiverShardID: mbHeader.ReceiverShardID,
+			SenderShardID:   core.MetachainShardId,
+			TxCount:         mbHeader.TxCount,
+		}
+		shardMBHeaders = append(shardMBHeaders, shardMBHdr)
+	}
+
+	s.miniBlocksSyncer.ClearFields()
+	err := s.miniBlocksSyncer.SyncPendingMiniBlocks(shardMBHeaders, timeToWait)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	peerMiniBlocks, err := s.miniBlocksSyncer.GetMiniBlocks()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	epochValidators := &sharding.EpochValidators{
+		EligibleValidators: make(map[string][]*sharding.SerializableValidator),
+		WaitingValidators:  make(map[string][]*sharding.SerializableValidator),
+	}
+
+	selfShardId := core.AllShardId
+	found := false
+	shouldSearchSelfId := len(publicKey) == 0
+	for _, mb := range peerMiniBlocks {
```

Comment:
This should have been included in a peer processor

Reply:
speak to the owner.

Reply:
:D